types/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+pub use crate::data_types::{Column, DataTypeNode};
+use crate::decoders::{ensure_size, read_string};
+use crate::error::TypesError;
+pub use crate::leb128::put_leb128;
+pub use crate::leb128::read_leb128;
+use bytes::BufMut;
+
+pub mod data_types;
+pub mod decoders;
+pub mod error;
+pub mod leb128;
+
+pub fn parse_rbwnat_columns_header(buffer: &mut &[u8]) -> Result<Vec<Column>, TypesError> {
```

Comment:
Actually, for a public API, this is a very specific and uncommon API (I mean `&mut &[u8]` instead of `impl Buf`). It's fine for internal code, but slightly strange for public one.

Reply:
We probably can leave it as-is for `v0.1.0`, because we can break this crate's code much often (if the `clickhouse` crate doesn't use it in *public* api)